File: 1302 Deepest Leaves Sum.go
func deepestLeavesSum(root *TreeNode) int {
    _, x := get(root, 0, 0, 0)
    return x
}

func get(root *TreeNode, level, highestLevel, sum int) (int, int) {
    if  root.Left != nil {
        highestLevel, sum = get(root.Left, level+1, highestLevel, sum)
    }
    
    if root.Right != nil {
        highestLevel, sum = get(root.Right, level+1, highestLevel, sum)
    }
    
    if root.Left == nil && root.Right == nil {
        if level > highestLevel {
            highestLevel = level
            sum = root.Val
        } else if level == highestLevel {
            sum += root.Val
        }
    }
    
    return highestLevel, sum
}

// Runtime: 28 ms, faster than 81.72% of Go online submissions for Deepest Leaves Sum.
// Memory Usage: 7.3 MB, less than 40.86% of Go online submissions for Deepest Leaves Sum.

File: 1302 Deepest Leaves Sum.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    private int maxLevel;
    private int sum;
    public int DeepestLeavesSum(TreeNode root) {
        Traverse(root, 0);
        return sum;
    }
    
    private void Traverse(TreeNode node, int level)
    {
        if(node == null) return;
        Traverse(node.left, level+1);
        if(level > maxLevel)
        {
            maxLevel = level;
            sum = node.val;
        }
        else if(level == maxLevel)
            sum += node.val;
        Traverse(node.right, level+1);
    }
}


Runtime: 120 ms, faster than 54.97% of C# online submissions for Deepest Leaves Sum.
Memory Usage: 34.9 MB, less than 9.94% of C# online submissions for Deepest Leaves Sum.
